Flatiron - Tuples chapter.md

```diff
@@ -0,0 +1,115 @@
+
+##What's a tuple good for?
+Now you're probably wondering why on earth you would want to use a tuple and, if that's where your mind has gone, I can't blame you.  Rest assured that there *are* some circumstances where a tuple is extremely useful, kind of like that weird tool you walk by in the hardware store that twists your mind but, in the right hands and in the right situation, can get the job done like nothing else.
+
+Tuples are the tool of choice when you want to pass highly structured data *to* a function or return highly structured data *from* a function but don't need the additional capabilities of a full-blown class or struct and don't want to go through the trouble of creating one.
+
+The most common time you'll find yourself using a tuple is iterating over a dictionary.  Iterating over an array returns *one* thing (an element) for each pass of the iteration, but iterating over a dictionary returns *two* things for each pass (namely the key and the value) .  Therefore, the iteration returns a tuple to you (with the key and value components correctly typed to how the dictionary was declared).
+
+**iteration**
+```
+// iterate over an array
+for name in list {
+	print(name)
+}
+"Darth"
+"Ben"
+"Luke"
+
+// iterate over a Dictionary (note the tuple)
+for (name, role) in simpleDictionary {
+	print(name + "is the " + role + " in Star Wars")
+}
+Luke is the protagonist in Star Wars
+```
+
+**Returning multiple values from functions**
+You can use a tuple to easily return related data from a function.  For example, here's a function that shows how you can, in a single line of code, return useful related information about a series of numbers that otherwise would take *four* lines of code:
+```
+func **dataSetStuff**( data: [Double] ) -> (**smallest** :Double, **largest** :Double, **average** :Double, **median** :Double) { 
+	var smallest = 0.0
+	var largest = 0.0
+	var average = 0.0
+	var median = 0.0
+	...  // compute the values and assign them appropriately
+	return (smallest, largest, average, median)
+	 }
+You call the function and **use the returned tuple** like this:
+let numbers = [ 23.0, 66.38, 35.512, 83.2, 41.89 ]
+let r = dataSetStuff( numbers )
+print( r.smallest )
+23.0
+print( "average: " + r.average ", ", median: " + r.median )
+average: 50.0, median: 41.89
+You **create a tuple** like this:
+let simpleTuple = ( 34, "who?", gamma:[1, 4, 16 ] )
+print( simpleTuple.0 )
+34
+```
+
+You create a tuple with **property names** like this:
+```
+// note how we're specifying keys for each value in the tuple
+let namedTuple = ( alpha:34, beta:"who?", gamma:[1, 4, 16 ] )
+print( namedTuple.alpha )
+34
+print (namedTuple.0 )  // you can STILL use idices to access if you like
+34
+```
+
+You can also change the value assigned to any index or key after creating the tuple (you just can't change the keys or types of each element).
+```
```

Comment:
Feel free to show them what this would look like in code (meaning the can't change it to a different type)

![PersonExample](http://i.imgur.com/cPwujYa.png)
Notice when I type a period after the `person` tuple to access its elements, it auto-completes showing me the names of the elements as well as their type (which they are both of type `String`), this means I can't change one to be an `Int`, otherwise I would be met with the following error:

![error](http://i.imgur.com/pGLXf1A.png)